provider/ec2/image.go

```diff
@@ -44,20 +37,53 @@ func filterImages(images []*imagemetadata.ImageMetadata, ic *instances.InstanceC
 
 // findInstanceSpec returns an InstanceSpec satisfying the supplied instanceConstraint.
 func findInstanceSpec(
+	controller bool,
 	allImageMetadata []*imagemetadata.ImageMetadata,
+	instanceTypes []instances.InstanceType,
 	ic *instances.InstanceConstraint,
 ) (*instances.InstanceSpec, error) {
 	logger.Debugf("received %d image(s)", len(allImageMetadata))
-	// If the instance type is set, don't also set a default CPU power
-	// as this is implied.
-	cons := ic.Constraints
-	if cons.CpuPower == nil && (cons.InstanceType == nil || *cons.InstanceType == "") {
-		ic.Constraints.CpuPower = instances.CpuPower(defaultCpuPower)
+	if controller {
+		ic.Constraints = withDefaultControllerConstraints(ic.Constraints)
+	} else {
+		ic.Constraints = withDefaultNonControllerConstraints(ic.Constraints)
 	}
 	suitableImages := filterImages(allImageMetadata, ic)
 	logger.Debugf("found %d suitable image(s)", len(suitableImages))
 	images := instances.ImageMetadataToImages(suitableImages)
-
-	instanceTypes := ec2instancetypes.RegionInstanceTypes(ic.Region)
 	return instances.FindInstanceSpec(images, ic, instanceTypes)
 }
+
+// withDefaultControllerConstraints returns the given constraints,
+// updated to choose a default instance type appropriate for a
+// controller machine. We use this only if the user does not specify
+// any constraints that would otherwise control the instance type
+// selection.
+//
+// At the time of writing, this will choose
+//   - t2.medium, for VPC
```

Comment:
This is just a default. If you're expecting to manage a large environment, then you should specify a higher minimum.

We did talk about specifying an instance type by name, but the problem with that is that when T3 inevitably comes along, we won't select it without further code changes. We don't yet auto-update instance types, but my recent changes should make it fairly easy to do so.